daemon/daemon.go

```diff
@@ -951,6 +959,13 @@ func (daemon *Daemon) Shutdown() error {
 // Mount sets container.BaseFS
 // (is it not set coming in? why is it unset?)
 func (daemon *Daemon) Mount(container *container.Container) error {
+	daemon.mountingContainers++
```

Comment:
It could be solved:

If `ShutDown` is called, when checking `daemon.mountingContainers==0` the daemon shutdown flag is set to true. `Mount` will return directly with error "daemon is shutting down".

If `Mount` could go on mounting rootfs, the shutting down flag must be false. `cleanup` is not called yet, and will check the counter later.


And I will change to use atomic counter.